src/Service/Api.php

```diff
@@ -1798,13 +1799,35 @@ public function getAutoscalingSettingsLink(Environment $environment, bool $manag
     public function getAutoscalingSettings(Environment $environment)
     {
         try {
-            $settings = $environment->getAutoscalingSettings();
+            $result = $environment->runOperation('autoscaling', 'get');
+        } catch (EnvironmentStateException $e) {
+            if ($e->getEnvironment()->status === 'inactive') {
+                throw new EnvironmentStateException('The environment is inactive', $e->getEnvironment());
+            }
+            throw $e;
+        }
+        return new AutoscalingSettings($result->getData(), null, $environment->client);
```

Comment:
Hm

`->client` can't be used here

I think the previous `$environment->getAutoscalingSettings()` was clearer and it means the object returned has the base URL and client set up correctly.